snap/implicit.go

```diff
@@ -139,6 +139,11 @@ func addImplicitHooksFromContainer(snapInfo *Info, snapf Container) error {
 }
 
 func addHookName(snapInfo *Info, hookName string) {
+	// Verify that the hook name is actually supported. If not, ignore it.
+	if !IsHookSupported(hookName) {
```

Comment:
Alternatively, rename the function to make the case more clear: `addHookIfValid` or something.